library/block/block_entity/src/main/java/org/quiltmc/qsl/block/entity/api/QuiltBlockEntity.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.entity.api;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.world.World;
+
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+/**
+ * Extensions of {@link BlockEntity}, provides some utilities like easy syncing.
+ */
+@InjectedInterface(BlockEntity.class)
+public interface QuiltBlockEntity {
+	/**
+	 * Attempts to synchronize the block entity data to the client.
+	 *
+	 * @throws IllegalStateException if called on the logical client
+	 * @throws NullPointerException  if there's no world associated with the block entity
+	 */
+	default void sync() {
+		if (this instanceof BlockEntity blockEntity) {
+			World world = blockEntity.getWorld();
+
+			Preconditions.checkNotNull(world); // Maintain distinct failure case from below.
```

Comment:
Not sure what qsl typically does, but "vanilla" jdk `Objects.requireNonNull` might be a bit nicer